components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1288,6 +1291,16 @@ private String handleFailureAuthorization(OAuthMessage oAuthMessage, OIDCSession
         return EndpointUtil.getErrorRedirectURL(oAuthMessage.getRequest(), oauthProblemException, oauth2Params);
     }
 
+    private String handleFailureBeforeConsent(OAuthMessage oAuthMessage, OAuth2Parameters oauth2Params,
```

Comment:
Shall we change this method name to better one, like handleAuthorizationFailureBeforeConsent as it would be better if you add which failure is going to be handled from this method.